### PR TITLE
Fix undefined array_key

### LINKED ISSUE #10587: [Bug]: Undefined array_key
### Expected behavior

Pimcore works

### Actual behavior

Pimcore doesnt work

### Steps to reproduce

1. Install plain pimcore 10.2
2. Open pimcore
3. Error, Undefined array_key 'default'

### PR DESCRIPTION
**Pull request by dpfahlbusch**
## Changes in this pull request  
Resolves #10587

### CODE REVIEW

File: lib/Perspective/Config.php [hunk 1]
```diff
@@ -253,11 +253,19 @@ public static function getRuntimePerspective(\Pimcore\Model\User $currentUser =
     protected static function getRuntimeElementTreeConfig($name)
     {
         $masterConfig = self::get()->toArray();
+        
+        if (!array_key_exists($name, $masterConfig)) {
+            return [];
+        }
 
         $config = $masterConfig[$name];
```
Comment (jdreesen) [inline lib/Perspective/Config.php]:
```suggestion
        
        $config = $masterConfig[$name] ?? null;
```

File: lib/Perspective/Config.php [hunk 2]
```diff
@@ -253,11 +253,19 @@ public static function getRuntimePerspective(\Pimcore\Model\User $currentUser =
...
         if (!$config) {
             $config = [];
         }
+        
+        if (!array_key_exists('elementTree', $config)) {
+            return [];
+        }
```
Comment (jdreesen) [inline lib/Perspective/Config.php]: Not sure that this is right, as `$config['elementTree']` seems to be optional. 

Maybe get it like this `$tmpResult = $config['elementTree'] ?? [];` and skip the following `if (is_null($tmpResult)) {...}` (sorry, GitHub does not allow me to create a suggestion for this)

Comment (dpfahlbusch) [reply lib/Perspective/Config.php]: Sounds good, thank you :)

File: lib/Perspective/Config.php [hunk 3]
```diff
@@ -253,16 +253,18 @@ public static function getRuntimePerspective(\Pimcore\Model\User $currentUser =
...
             $config = [];
         }
-
-        $tmpResult = $config['elementTree'];
-        if (is_null($tmpResult)) {
-            $tmpResult = [];
+        
+        if (!array_key_exists('elementTree', $config)) {
+            return [];
         }
```
Comment (jdreesen) [inline lib/Perspective/Config.php]: This is not necessary anymore (and probably wrong anyways)
```suggestion
        
        if (!array_key_exists('elementTree', $config)) {
            return [];
        }
```

Comment (dpfahlbusch) [reply lib/Perspective/Config.php]: GitHub-Web messed it up ^^ should be fixed now :)

**Comment by dvesh3** [comment]
@dpfahlbusch thanks for your PR! However i'm afraid this is not an optimal solution as it still does not load the default config if none available from legacy php file, settings-store and symfony container. I would just simply load standard perspective config if none found here https://github.com/pimcore/pimcore/blob/10.x/lib/Perspective/Config.php#L85
```
if (!count($config)) {
    $config = static::getStandardPerspective();
}
```
Note: Nonexistent perspective requested by the user is handled in the event listener https://github.com/pimcore/pimcore/blob/10.x/bundles/AdminBundle/EventListener/UserPerspectiveListener.php#L89

**Comment by dpfahlbusch** [comment]
@dvesh3 hope this is the right way :)

**Comment by dvesh3** [comment]
@dpfahlbusch I would just use standard config in `get()` method, as this called at every step of loading perspectives. please see my PR https://github.com/pimcore/pimcore/pull/10591

**Comment by dvesh3** [comment]
closed in favor of #10591

**Pull request closed without merging**